src/main/java/frc/team2412/robot/Controls.java

```diff
@@ -106,4 +102,26 @@ private void bindLauncherControls() {
 						LauncherSubsystem.SPEAKER_SHOOT_SPEED_RPM,
 						LauncherSubsystem.SUBWOOFER_AIM_ANGLE));
 	}
+
+	// intake controls
+	private void bindIntakeControls() {
+		// later on set default command to check sensing notes
+		CommandScheduler.getInstance()
+				.setDefaultCommand(s.intakeSubsystem, new IntakeStopCommand(s.intakeSubsystem));
+		codriveIntakeInButton.onTrue(
+				Commands.parallel(
+						new IntakeInCommand(s.intakeSubsystem),
+						new IndexInCommand(s.intakeSubsystem),
+						new FeederInCommand(s.intakeSubsystem)));
+		codriveIntakeStopButton.onTrue(
+				Commands.parallel(
+						new IntakeStopCommand(s.intakeSubsystem),
+						new IndexStopCommand(s.intakeSubsystem),
+						new FeederStopCommand(s.intakeSubsystem)));
+		codriveIntakeSpitButton.onTrue(
+				Commands.parallel(
+						new IntakeReverseCommand(s.intakeSubsystem),
+						new IndexReverseCommand(s.intakeSubsystem),
+						new FeederReverseCommand(s.intakeSubsystem)));
```

Comment:
These will all crash since there's this check in the ParallelCommandGroup each of these create:
From https://github.com/wpilibsuite/allwpilib/blob/6afff996401343c3bfcc46a40a40abda429e5077/wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/ParallelCommandGroup.java#L51C1-L54C8
```
      if (!Collections.disjoint(command.getRequirements(), m_requirements)) {
        throw new IllegalArgumentException(
            "Multiple commands in a parallel composition cannot require the same subsystems");
      }
```

Reply:
For these you likely need to add a new command that does the combination that's needed. And to address the proliferation of tiny command classes that just call a method or two on a subsystem, consider adding command helpers to the subsystem (like DrivebaseSubsystem::rotateToAngle did) rather than small Command classes.